pkg/graveler/graveler.go

```diff
@@ -3109,6 +3164,14 @@ func (g *Graveler) Diff(ctx context.Context, repository *RepositoryRecord, left,
 		leftValueIterator.Close()
 		return nil, err
 	}
+	if rightBranch.CompactedBaseMetaRangeID != "" {
+		compactedDiffIterator, err := g.CommittedManager.Diff(ctx, repository.StorageNamespace, leftCommit.MetaRangeID, rightBranch.CompactedBaseMetaRangeID)
```

Comment:
`leftCommit` & `rightBranch.CompactedBaseMetaRangeID`? Can you explain why this is true?

Reply:
In case there's no compaction data on the right we return a combined iterator with:
1) diff iterator between the left committed data and the right committed data
2) iterator of committed data in the left branch
3) iterator of the staging data of the right branch

So in case there's compacted data on the right branch we need to change the first iterator to contain also changes that have already compacted to the right branch

Reply:
> In case there's no compaction data on the right we return a combined iterator with:

But this path handles the case where `rightBranch.CompactedBaseMetaRangeID != ""`

Reply:
So, why isn't it `JoinedDiff(stagingIterator, compactedDiffIterator)`? Isn't `iterator of committed data in the left branch` already being diffed in `compactedDiffIterator`?